### ISSUE TITLE
json_parse() corrupts strings that start with "@", converting them to 0.

### ISSUE THREAD

**Issue by BscotchSeth** [YoYoGames/GameMaker-Bugs#4727]
### Description

In runtime 2024.200.0.511, this test will fail:

```
var _some_json = { text : "@whatever" };
var _some_json_string = json_stringify(_some_json);
var _some_json_string_parsed = json_parse(_some_json_string);

if (_some_json.text != _some_json_string_parsed.text) {
    show_debug_message("Parsed text: " + string(_some_json_string_parsed.text));
    show_error("Parsed string became corrupted.", true);
}
```

The string `"@whatever"` becomes `0` when json_parsed.

I can attach a test project if necessary, but the test above is the entirety of the bug, so you can run it wherever/however you like.

### Expected Change

json_parse() should not corrupt the contents of the data it is parsing.

### Steps To Reproduce

1. Start GameMaker
2. Run the test above
3. Observe crash

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.200 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by adam-coster** [comment]
Probably has something to do with how they encode nonstandard JSON values (like `infinity`), since they represent those as `@`-prefixed strings.

**Comment by shichen85** [comment]
> Probably has something to do with how they encode nonstandard JSON values (like `infinity`), since they represent those as `@`-prefixed strings.
> 
> @shichen85 we should have our test suite include all of those specially encoded values, to make sure they encode/decode as expected.

OK, I am adding these ones to the test https://manual.gamemaker.io/monthly/en/#t=GameMaker_Language%2FGML_Overview%2FVariables%2FConstants.htm

**Comment by rwkay** [comment]
OK I have put in a fix for this... thanks for catching it.. due to code being rearranged to add the filter functions and I forgot to add the default case to an `if...else if....else if....` it should have had a final `else`.

**Comment by rwkay** [comment]
Fixed in 2024.2 - should be in the next Beta

**Comment by sihammill** [comment]
Fix verified in IDE v2024.200.0.497  Runtime v2024.200.0.514
